Recursor: unused lambda capture in unit tests

- Program: Recursor
 - Issue type: Bug report/Feature request <!-- delete the one that does not apply -->

### Short description
Compiler warning when compiling `test-syncres_cc8.cc`

### Environment
<!-- Tell us about the environment -->
 - Operating system: Arch Linux
 - Software version: git (d26856a32e7506817c291c49d8eb5a60129dd751)
 - Software source: git

### Steps to reproduce
```
autoreconf -fi
CC=clang CXX=clang++ LD=lld ./configure --enable-unit-tests
make
```

### Expected behaviour
No warnings.

### Actual behaviour
```
test-syncres_cc8.cc:701:51: warning: lambda capture 'fixedNow' is not used [-Wunused-lambda-capture]
  sr->setAsyncCallback([target,&queriesCount,keys,fixedNow](const ComboAddress& ip, const DNSName& domain, int type, bool doTCP, bool sendRDQuery, int EDNS0Level, struct timeval* now, boost::optional<Netmask>& srcmask, boost::optional<const ResolveContext&> context, LWResult* res, bool* chained) {
                                                 ~^~~~~~~~
1 warning generated.
```

--- a/pdns/recursordist/test-syncres_cc8.cc
+++ b/pdns/recursordist/test-syncres_cc8.cc
@@ -698,7 +698,7 @@ BOOST_AUTO_TEST_CASE(test_dnssec_rrsig_negcache_bogus_validity) {
   size_t queriesCount = 0;
   const time_t fixedNow = sr->getNow().tv_sec;
 
-  sr->setAsyncCallback([target,&queriesCount,keys,fixedNow](const ComboAddress& ip, const DNSName& domain, int type, bool doTCP, bool sendRDQuery, int EDNS0Level, struct timeval* now, boost::optional<Netmask>& srcmask, boost::optional<const ResolveContext&> context, LWResult* res, bool* chained) {
+  sr->setAsyncCallback([&queriesCount,keys](const ComboAddress& ip, const DNSName& domain, int type, bool doTCP, bool sendRDQuery, int EDNS0Level, struct timeval* now, boost::optional<Netmask>& srcmask, boost::optional<const ResolveContext&> context, LWResult* res, bool* chained) {
       queriesCount++;
 
       DNSName auth = domain;

